### PR TITLE
getting elemental changes on cluster provisioning back to rancher dashboard

### PR DESCRIPTION
**Pull request by aalves08**
Fixes https://github.com/rancher/elemental-ui/issues/50

- getting elemental changes on cluster provisioning back to rancher dashboard

### CODE REVIEW

**Comment by richard-cox** [comment]
@mantis-toboggan-md has done some great work to make `machine-config` available generically via pkgs in https://github.com/rancher/dashboard/pull/7207, it would be worth waiting for that to merge and updating

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 1]
```diff
@@ -1096,6 +1125,10 @@ export default {
           entry.config = await entry.config.save();
         }
 
+        if ( !entry.pool.hostnamePrefix ) {
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:1128]: This is a change in behaviour for most providers (or at least for DO). 

I couldn't find anywhere it's used (set or applied) in the dashboard apart from an RKE1 use case. In `elemental-ui` there  were only references  to here.  This is definitely needed right? If so, as much as it pains me, we'll need another isElementalCluster

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue (hunk 1 above)
Comment (nwmac) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:1128]: @aalves08 Is this used? Created a normal RKE2 cluster and was expecting this to appear somewhere in the cluster object or machine pools, but don't see it being persisted.

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 2]
```diff
@@ -556,11 +561,17 @@ export default {
     },
 
     machineConfigSchema() {
+      let schemaAddress;
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue]: Done - changed to schema and used direct return of schema object

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue (hunk 1 above)
Comment (nwmac) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue]: Added gate on Elemental cluster

**Comment by nwmac** [comment]
@richard-cox Updated as discussed: https://github.com/rancher/dashboard/pull/7543/commits/051c150b986ed6a00cf53260cff8186fbe18e219

**Review by richard-cox — APPROVED** [review]

**Pull request merged**